commit bf7f22b9cac74a1e3d8b8e77350db2baca2c35be
Author: Jason Gaston <jason.d.gaston@intel.com>
Date:   Mon Jan 28 17:36:45 2008 -0800

    ata_piix: IDE mode SATA patch for Intel ICH10 DeviceID's
    
    This patch adds the Intel ICH10 IDE mode SATA Controller DeviceID's.
    
    Signed-off-by:  Jason Gaston <jason.d.gaston@intel.com>
    Signed-off-by: Jeff Garzik <jeff@garzik.org>

Adjusted to apply to Debian's 2.6.24 by dann frazier <dannf@hp.com>

diff -urpN linux-source-2.6.24.orig/drivers/ata/ata_piix.c linux-source-2.6.24/drivers/ata/ata_piix.c
--- linux-source-2.6.24.orig/drivers/ata/ata_piix.c	2008-04-18 11:48:23.000000000 -0600
+++ linux-source-2.6.24/drivers/ata/ata_piix.c	2008-05-01 17:38:14.000000000 -0600
@@ -221,6 +221,14 @@ static const struct pci_device_id piix_p
 	{ 0x8086, 0x292e, PCI_ANY_ID, PCI_ANY_ID, 0, 0, ich8_sata_ahci },
 	/* SATA Controller IDE (Tolapai) */
 	{ 0x8086, 0x5028, PCI_ANY_ID, PCI_ANY_ID, 0, 0, tolapai_sata_ahci },
+	/* SATA Controller IDE (ICH10) */
+	{ 0x8086, 0x3a00, PCI_ANY_ID, PCI_ANY_ID, 0, 0, ich8_sata_ahci },
+	/* SATA Controller IDE (ICH10) */
+	{ 0x8086, 0x3a06, PCI_ANY_ID, PCI_ANY_ID, 0, 0, ich8_2port_sata },
+	/* SATA Controller IDE (ICH10) */
+	{ 0x8086, 0x3a20, PCI_ANY_ID, PCI_ANY_ID, 0, 0, ich8_sata_ahci },
+	/* SATA Controller IDE (ICH10) */
+	{ 0x8086, 0x3a26, PCI_ANY_ID, PCI_ANY_ID, 0, 0, ich8_2port_sata },
 
 	{ }	/* terminate list */
 };
